samples/packages/asset-import/src/main/kotlin/com/atlan/pkg/aim/DomainImporter.kt

```diff
@@ -0,0 +1,157 @@
+/* SPDX-License-Identifier: Apache-2.0
+   Copyright 2023 Atlan Pte. Ltd. */
+package com.atlan.pkg.aim
+
+import com.atlan.model.assets.Asset
+import com.atlan.model.assets.DataDomain
+import com.atlan.model.assets.DataProduct
+import com.atlan.model.fields.AtlanField
+import com.atlan.pkg.cache.DataDomainCache
+import com.atlan.pkg.serde.RowDeserializer
+import com.atlan.pkg.serde.cell.DataDomainXformer.DATA_DOMAIN_DELIMITER
+import com.atlan.pkg.serde.csv.CSVImporter
+import com.atlan.pkg.serde.csv.ImportResults
+import mu.KotlinLogging
+import java.util.concurrent.atomic.AtomicInteger
+import kotlin.math.max
+import kotlin.text.Regex
+
+/**
+ * Import data domains (only) into Atlan from a provided CSV file.
+ *
+ * Only the data domains and attributes in the provided CSV file will attempt to be loaded.
+ * By default, any blank values in a cell in the CSV file will be ignored. If you would like any
+ * particular column's blank values to actually overwrite (i.e. remove) existing values for that
+ * asset in Atlan, then add that column's field to getAttributesToOverwrite.
+ *
+ * @param filename name of the file to import
+ * @param attrsToOverwrite list of fields that should be overwritten in Atlan, if their value is empty in the CSV
+ * @param updateOnly if true, only update an asset (first check it exists), if false allow upserts (create if it does not exist)
+ * @param batchSize maximum number of records to save per API request
+ * @param fieldSeparator character to use to separate fields (for example ',' or ';')
+ */
+class DomainImporter(
+    private val filename: String,
+    private val attrsToOverwrite: List<AtlanField>,
+    private val updateOnly: Boolean,
+    private val batchSize: Int,
+    private val fieldSeparator: Char,
+) : CSVImporter(
+    filename,
+    logger = KotlinLogging.logger {},
+    typeNameFilter = DataDomain.TYPE_NAME,
+    attrsToOverwrite,
+    updateOnly = updateOnly,
+    batchSize = batchSize,
+    trackBatches = true, // Always track batches for GTC importers, to ensure cache is managed
+    fieldSeparator = fieldSeparator,
+) {
+    private var levelToProcess = 0
+
+    // Maximum depth of any domain in the CSV -- will be updated on first pass through the CSV
+    // file by includeRow() method
+    private val maxCategoryDepth = AtomicInteger(1)
```

Comment:
Change variables / comments regarding `category` to `domain`

Reply:
done